Sources/performance-worker/BenchmarkService.swift

```diff
@@ -0,0 +1,135 @@
+        if message.responseStatus.isInitialized {
+          try self.echoStatus(responseStatus: message.responseStatus)
+        }
+        try await writer.write(
+          Grpc_Testing_BenchmarkService.Method.StreamingCall.Output.with {
+            $0.payload = message.payload
+          }
+        )
+      }
+      return [:]
+    }
+  }
+
+  /// Single-sided unbounded streaming from client to server
+  /// The server returns the client payload as-is once the client does WritesDone
+  func streamingFromClient(
+    request: ServerRequest.Stream<Grpc_Testing_BenchmarkService.Method.StreamingFromClient.Input>
+  ) async throws
+    -> ServerResponse.Single<Grpc_Testing_BenchmarkService.Method.StreamingFromClient.Output>
+  {
+    throw RPCError(code: .unimplemented, message: "The RPC is not implemented.")
+  }
+
+  /// Single-sided unbounded streaming from server to client
+  /// The server repeatedly returns the client payload as-is
+  func streamingFromServer(
+    request: ServerRequest.Single<Grpc_Testing_BenchmarkService.Method.StreamingFromServer.Input>
+  ) async throws
+    -> ServerResponse.Stream<Grpc_Testing_BenchmarkService.Method.StreamingFromServer.Output>
+  {
+    return ServerResponse.Stream { writer in
+      if request.message.responseStatus.isInitialized {
+        try self.echoStatus(responseStatus: request.message.responseStatus)
+      }
+
+      while true {
+        try await writer.write(
+          Grpc_Testing_BenchmarkService.Method.StreamingCall.Output.with {
+            $0.payload = request.message.payload
+          }
+        )
+        try await Task.sleep(nanoseconds: 10)
+      }
+    }
+  }
+
+  /// Two-sided unbounded streaming between server to client.
+  /// Both sides send the content of their own choice to the other.
+  func streamingBothWays(
+    request: GRPCCore.ServerRequest.Stream<
+      Grpc_Testing_BenchmarkService.Method.StreamingBothWays.Input
+    >
+  ) async throws
+    -> GRPCCore.ServerResponse.Stream<Grpc_Testing_BenchmarkService.Method.StreamingBothWays.Output>
+  {
+    return ServerResponse.Stream { writer in
+      while true {
+        try await writer.write(
+          Grpc_Testing_BenchmarkService.Method.StreamingCall.Output.with {
+            $0.payload = Grpc_Testing_Payload.with {
+              $0.body = Data(count: Int.random(in: 1 ..< 10))
+            }
+          }
+        )
+        try await Task.sleep(nanoseconds: 10)
+      }
+    }
+  }
+}
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+extension BenchmarkService {
+  private func echoStatus(responseStatus: Grpc_Testing_EchoStatus) throws {
+    guard let code = Status.Code(rawValue: Int(responseStatus.code))
+    else {
+      throw RPCError(code: .invalidArgument, message: "The response status code is invalid.")
+    }
+    let status = Status(code: code, message: responseStatus.message)
+    if let error = RPCError(status: status) {
+      throw error
+    }
```

Comment:
nit: we always create the `Status` but don't always use it (in fact we don't need to use it at all):

```swift
if let code = RPCError.Code(code) {
  throw RPCError(code: code, message: "...")
}
```